src/vs/workbench/contrib/terminal/browser/remotePty.ts

```diff
@@ -93,17 +91,18 @@ export class RemotePty extends Disposable implements ITerminalChildProcess {
 		}
 
 		this._startBarrier.wait().then(_ => {
-			this._remoteTerminalChannel.input(this._id, data);
+			this._remoteTerminalChannel.input(this.id, data);
 		});
 	}
 
 	resize(cols: number, rows: number): void {
-		if (this._inReplay) {
+		if (this._inReplay || this._lastDimensions.cols === cols && this._lastDimensions.rows === rows) {
```

Comment:
hmm, will keep my eye out for issues, i wonder if this could go wrong

Reply:
Since there should only ever be a single window attached, it should stay in sync